Makefile

```diff
@@ -0,0 +1,201 @@
+SHELL := /usr/bin/env bash -eo pipefail
+MKFILE_DIR := $(abspath $(dir $(abspath $(firstword $(MAKEFILE_LIST)))))
+
+# NOTE: variables that can be defined/overwritten
+#ENV_DIR or ENV
+
+# NOTE: internal variables
+# Please ignore the following line if you're not a Wire employee
+ENVIRONMENTS_DIR := $(abspath $(CURDIR)/../cailleach/environments)
+
+
+
+ifndef ENV_DIR
+ifndef ENV
+$(error "[ERR] Undefined variable: Please define either ENV or ENV_DIR")
+else
+ENV_DIR = $(ENVIRONMENTS_DIR)/$(ENV)
+endif
+endif
+
+
+
+######################################## TERRAFORM #############################
+export TF_DATA_DIR = $(ENV_DIR)/.terraform
+TERRAFORM_WORKING_DIR = $(MKFILE_DIR)/terraform/environment
+
+
+# NOTE: leverage make's ability to noop if target already exists, but
+#       this also means that re-init must be triggered explicitly
+$(TF_DATA_DIR):
+	cd $(TERRAFORM_WORKING_DIR) \
+	&& terraform init -backend-config=$(ENV_DIR)/backend.tfvars
+
+.PHONY: re-init
+re-init: check-inputs-terraform
+	make --always-make $(TF_DATA_DIR)
+
+# NOTE: alias for 're-init'
+.PHONY: init
+.SILENT: init
+init:
+init:
+	echo "[WARN] 'make init' is deprecated. Please use re-init moving forward"
```

Comment:
Why keep deprecated things in a new file? Why not just delete this target?

Reply:
I'm fine with that too. I tried to migrate the *interfaces* as they were and only get rid of the redundant stuff. I wanted to make changing CICD scripts a little bit less error-prone. I'll remove `make init`. We then just have to be a bit more conscious when making the changes there.